spark/src/main/java/org/apache/iceberg/spark/source/BaseDataReader.java

```diff
@@ -125,11 +126,13 @@ public void close() throws IOException {
 
   protected InputFile getInputFile(FileScanTask task) {
     Preconditions.checkArgument(!task.isDataTask(), "Invalid task type");
-    return inputFiles.get(task.file().path().toString());
+    return getInputFile(task.file().path().toString());
```

Comment:
probably best to path().toURI() for best guarantee of handling of spaces

Reply:
This isn't a Hadoop `Path`, it is a `CharSequence`. That's how we generally avoid issues like encoding.